guide/build-config.md

```diff
@@ -138,13 +139,17 @@ builds:
   scmUrl: https://<external git server>/<repo>.git
 ```
 
-#### environmentId v/s systemImageId
+#### environmentId v/s environmentName v/s systemImageId
 
-The build environment used for builds has a unique id and system image id.
+The build environment used for builds has a unique id, name and system image id.
 
-We allow the user to specify the build environment using either `environmentId` or the `systemImageId`. `environmentId` refers to the exact id of the build environment as stored in the PNC database. This is what we traditionally use, but it is problematic since the ids are different between PNC servers (Devel, Stage, Prod), which makes the `build-config.yaml` not very portable between PNC servers.
+We allow the user to specify the build environment using either `environmentId`, `environmentName` or the `systemImageId`. `environmentId` refers to the exact id of the build environment as stored in the PNC database. This is what we traditionally use, but it is problematic since the ids are different between PNC servers (Devel, Stage, Prod), which makes the `build-config.yaml` not very portable between PNC servers.
 
-The `systemImageId` in contrast, is the same between different PNC servers and should be used if portability is a concern. To get the `environmentId` or the `systemImageId` of the build image, simply run:
+The `environmentName` in contrast, is the same between different PNC servers and should be used if portability is a concern. 
+
+The `systemImageId` is mostly the same between different PNC servers, the difference to `environmentName` is that the `systemImageId` **preserves** the docker-image version of the environment whereas environments specified by `environmentName` always use the **latest** updated version (**NOTE**: `systemImageId` with `:latest` suffix behaves the same way as `environmentName`).  
```

Comment:
`systemImageId` is also now deprecated right @vibe13 ?

Reply:
Huh. Haven't seen that mentioned anywhere.

Reply:
Yeah you're right about it not being mentioned anywhere. IIRC from a convo with Andrea, only Marek was using it for CPAAS. If no other users are using it, I think Andrea and I agreed to perhaps deprecate it. I'll wait for his reply on this subject.

Reply:
After a discussion with Andrea: 
- the fields environmentId and systemImageId won't be deprecated
- environmentName will be highly prefered way for users to specify environments

Reply:
Gotcha! :)